src/main.cpp

```diff
@@ -581,6 +583,51 @@ struct router_thread
 		}
 	}
 
+	void send_request(queued_node_t const& n)
+	{
+		char remote_ip[18];
+		size_t remote_ip_len = build_ip_field(n.ep, remote_ip);
+		std::array<char, 4> transaction_id = compute_tid(secret1, remote_ip, remote_ip_len);
+		bound_socket& sock = socks[n.sock_idx];
+
+		// send find_nodes to this node
+		bencoder b(response, sizeof(response));
+		b.open_dict();
+
+		// args dict
+		b.add_string("a");
+		b.open_dict();
+		b.add_string("id"); b.add_string(sock.node_id.data(), sock.node_id.size());
+		char random_target[20];
+		std::generate(random_target, random_target + 20, &std::rand);
+		b.add_string("target"); b.add_string(random_target, sizeof(random_target));
+		b.close_dict();
+
+		b.add_string("t"); b.add_string(transaction_id);
+		b.add_string("q"); b.add_string("find_node");
+
+		if (version[0] != 0)
+		{
+			b.add_string("v");
+			b.add_string(version.data(), 4);
+		}
+
+		b.add_string("y"); b.add_string("q");
+
+		b.close_dict();
```

Comment:
The request should have a "want" argument with both "n4" and "n6" listed so that it gets both IPv4 and IPv6 nodes.